### DISCUSSION TITLE
How do I get the list of all the available built-in mechanisms?

### DISCUSSION THREAD

**Discussion by lungsi** [arbor-sim/arbor#1587] [Q&A] [answered]
Hi,
I was wondering if I can view all the available built-in mechanisms?

The [documentation](https://docs.arbor-sim.org/en/latest/python/mechanisms.html) tells us accessing specific mechanisms as in
```
import arbor
cat = arbor.default_catalogue()
mech = cat['expsyn']
```
or checking if a mechanism exists with
```
if 'hh' in arbor.default_catalogue():
  print("Found HH mechanism.")
```

But these cases assumes that the user already knows the name of the mechanism. How can I view all the available built-in mechanisms?

**Comment by noraabiakar** [comment, marked as answer]
You should be able to iterate over the catalogue and get the names of the mechanisms: 
```
for mech in arbor.default_catalogue(): 
  print(mech)
```

**Reply by lungsi** [reply]
Thanks so much. It was silly of me not to figure it out :)

**Reply by noraabiakar** [reply]
No problem! :slightly_smiling_face:
